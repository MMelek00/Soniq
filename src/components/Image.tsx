import * as React from 'react';
import styled from 'styled-components/native';
import {getWindowHeight, getWindowWidth} from '../utils/layout';
type ImageProps = {
  width: number;
  height: number;
  url: string;
};

const StyledImage = styled.Image<{width: number; height: number}>`
  width: ${props => props.width}px;
  height: ${props => props.height}px;
  border-radius: 5px;
`;

const Image = ({width, height, url}: ImageProps) => {
  return (
    <StyledImage
      source={{uri: url}}
      width={getWindowHeight(width)}
      height={getWindowWidth(height)}
    />
  );
};

export default Image;
