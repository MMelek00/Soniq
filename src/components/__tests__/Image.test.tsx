import 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';
import Image from '../Image';

describe('<Image />', () => {
  it('renders correctly with Error', () => {
    const mockUrl = 'https://api.slingacademy.com/public/sample-photos/6.jpeg';
    const tree = renderer.create(
      <Image url={mockUrl} width={22} height={35} />,
    );
    expect(tree).toMatchSnapshot();
  });
});
