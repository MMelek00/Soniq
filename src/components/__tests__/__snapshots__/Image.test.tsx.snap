// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> renders correctly with Error 1`] = `
<Image
  height={262.5}
  source={
    {
      "uri": "https://api.slingacademy.com/public/sample-photos/6.jpeg",
    }
  }
  style={
    [
      {
        "borderRadius": 5,
        "height": 262.5,
        "width": 293.48,
      },
    ]
  }
  width={293.48}
/>
`;
